### PR TITLE
Refactor some VUID structs, helper codegen

### PR DESCRIPTION
**Pull request by mark-lunarg**
Thanks to Nate, added named-initialization to the VUID structs we pass around, and removed some debugger-unfriendly `#defines` from codegen.

Passed [internal CI](http://erusea:4040/#/c/7754/).

### CODE REVIEW

File: layers/core_validation_types.h [hunk 1]
```diff
@@ -635,7 +635,7 @@ struct RENDER_PASS_STATE : public BASE_NODE {
 };
 
 // Autogenerated as part of the vk_validation_error_message.h codegen
-enum CMD_TYPE { VUID_CMD_ENUM_LIST(CMD_) };
+///enum CMD_TYPE { VUID_CMD_ENUM_LIST(CMD_) };
```
Comment (ncesario-lunarg) [inline layers/core_validation_types.h:638]: Remove? If not, `//` instead of `///`?

**Review by ncesario-lunarg — APPROVED** [review]
👍

File: layers/core_validation_types.h (hunk 1 above)
Comment (mark-lunarg) [reply layers/core_validation_types.h:638]: Yup, just forgot to delete during cleanup -- thanks!

**Comment by mark-lunarg** [comment]
Gonna wait on #2269 (unless it takes too long!).

File: layers/core_validation_types.h (hunk 1 above)
Comment (sfricke-samsung) [reply layers/core_validation_types.h:638]: is there a commit not pushed removing this?

**Review by sfricke-samsung — APPROVED** [review]
Pulled down and lgtm! 

This is such a great improvement! Excited as now the VUID DrawDispatch table will actually be enjoyable to add to :smile:

**Comment by mark-lunarg** [comment]
Thanks Spencer!  This was Nate's implementation, mine was not nearly as good.

**Comment by mark-lunarg** [comment]
Nasty rebase.

**Pull request merged**